### ISSUE TITLE
Some library_html5.js canvas operations don't proxy to main thread

### ISSUE THREAD

**Issue by VirtualTim** [emscripten-core/emscripten#8203]
I found `emscripten_get_canvas_element_size_main_thread` and `emscripten_set_canvas_element_size_main_thread` were failing when run from a thread. There was code there to proxy those calls to the main thread, but it wasn't working.
I've tried to fix this two (see #8185), but I need to check for more cases, and fix the failing test.

I figured I'd just log this for visibility. I'll close it when the PR is merged.

**Comment by VirtualTim** [comment]
Simple Example:
```
#include <emscripten/html5.h>
int main(int argc, char** argv) {
  int w, h;
  emscripten_get_canvas_element_size("#canvas", &w, &h);
  return EXIT_SUCCESS;
}
```
Compile with: `emcc test.cpp -s USE_PTHREADS=1 -s PROXY_TO_PTHREAD=1 -s OFFSCREENCANVAS_SUPPORT=0 -o test.html`

**Comment by VirtualTim** [comment]
Fixed in #8185.
